app/services/find_status_change_audits.rb

```diff
@@ -0,0 +1,33 @@
+class FindStatusChangeAudits
+  attr_reader :application_choice
+
+  StatusChange = Struct.new(:status, :changed_at, :user)
+
+  def initialize(application_choice:)
+    @application_choice = application_choice
+  end
+
+  def call
+    status_audits.map { |audit| status_change_for(audit) }
+  end
+
+private
+
+  def status_audits
+    application_choice.audits.select do |audit|
+      status_change?(audit)
+    end
+  end
+
+  def status_change_for(audit)
+    StatusChange.new(
+      audit.audited_changes['status'][1],
+      audit.created_at,
+      audit.user || audit.username,
```

Comment:
I think falling back to `.username` will always cause this to result in "System" downstream? Maybe the calling code in the component should proactively check for a string and render that? Or perhaps we should supply a `User`-like thing in this position?

Reply:
I was seeing both Candidate and Provider attributed history items. However I got some odd results and it fell back to 'System' in dev whenever rails reloading the model classes because `audit.user.is_a? Candidate` returned false further downstream.

Reply:
I found that too when I spiked this. `user.is_a? Candidate` just didn't work when it came out of an audit. I ended up checking `if audit.user_type == "Candidate"`

Reply:
The mystery was pretty good — iirc `user.class.id` (ie the internal Ruby object id) and `Candidate.id` were identical and everything